provider/aws/aws.go

```diff
@@ -509,9 +528,30 @@ func (p *AWSProvider) submitChanges(ctx context.Context, changes []*route53.Chan
 	for z, cs := range changesByZone {
 		var failedUpdate bool
 
-		batchCs := batchChangeSet(cs, p.batchChangeSize)
+		// group changes into new changes and into changes that failed in a previous iteration and are retried
+		var newChanges, retriedChanges Route53Changes
+		for _, c := range cs {
+			found := false
+			if p.failedChangesQueue[z] != nil {
+				for _, failedChange := range p.failedChangesQueue[z] {
+					if c == failedChange {
+						retriedChanges = append(retriedChanges, c)
+						found = true
+					}
+				}
+				p.failedChangesQueue[z] = nil // clear the queue
+			}
+			if !found {
+				newChanges = append(newChanges, c)
+			}
+		}
+		batchCs := append(batchChangeSet(newChanges, p.batchChangeSize), batchChangeSet(retriedChanges, p.batchChangeSize)...)
 
 		for i, b := range batchCs {
+			if len(b) == 0 {
+				continue
+			}
```

Comment:
How come we need this now? Can `b` now be empty where it wasn't possible before? If the change above is just recording the `newChanges` and `failedChanges` the batch size should be the same as before.

Reply:
Ok, I think I know why we need it. `retriedChanges` can obviously be empty and so can `newChanges` if all desired records were part of `failedChangesQueue`, i.e. have been tried before.